Xamarin.PropertyEditing.Mac/Controls/PropertyEditorControl.cs

```diff
@@ -146,6 +112,66 @@ public sealed override void ViewDidChangeEffectiveAppearance ()
 
 			AppearanceChanged ();
 		}
+
+		public void OnNextResponderRequested ()
+		{
+			if (TableView != null) {
+				nint row = TableView.RowForView (this) + 1;
+
+				NSView view;
+				PropertyEditorControl ctrl = null;
+
+				for (; row < TableView.RowCount; row++) {
+					view = TableView.GetView (0, row, makeIfNecessary: false);
+					if (view is PropertyEditorControl pec) { // This is to include the CategoryContainer
+						ctrl = pec;
+					} else {
+						ctrl = (view as EditorContainer)?.EditorView?.NativeView as PropertyEditorControl;
+					}
+
+					if (ctrl?.viewModel != null && !ctrl.viewModel.IsInputEnabled) {
+						ctrl = null;
+					}
+
+					if (ctrl != null) {
+						Window?.MakeFirstResponder (ctrl.FirstKeyView);
+						return;
+					}
+				}
+			}
+		}
+
+		public void OnPreviousResponderRequested ()
```

Comment:
A lot of duplicated code between these, maybe they could be unified a bit? Maybe both call into `OnNextResponderRequested(bool reverse)` or something?

Reply:
The random type casts and inconsistent nullability propagation usages `?.` makes it hard to read/understand.

Reply:
Also, I agree with Sandy here, there's a lot of duplicated code that's slightly different and it's hard to follow.